File: Data Structures & Algorithms/reverse-a-linked-list/submission-0.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

class Solution {
    public ListNode reverseList(ListNode head) {
        if(head ==null) return head;
        ListNode node = head;
        Deque<ListNode> stack = new ArrayDeque<>();
        stack.push(node);
        while(node.next != null){
            stack.push(node.next);
            node=node.next;
        }
        boolean popped = false;
        ListNode ans = null;
        while(!stack.isEmpty()){
            ListNode n = stack.pop();
            if(!popped) ans = n;
            popped = true;
            n.next = stack.peek(); 
        }
        return ans;
    }
}

File: Data Structures & Algorithms/add-two-numbers/submission-0.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */

func addTwoNumbers(l1 *ListNode, l2 *ListNode) *ListNode {
	remainder := 0
	c1 := l1
	c2 := l2
	
	var head *ListNode;
	var prev *ListNode;

	for c1!= nil || c2 !=nil{
		l,r := 0,0
		if c1!=nil{
			l=c1.Val
			c1 = c1.Next
		}
		if c2!=nil{
			r=c2.Val
			c2=c2.Next
		}

		sum := l + r + remainder
		if sum > 9 {
			sum = sum-10
			remainder =1
		} else {
			remainder =0
		}
		this := &ListNode{Val: sum}
		if head == nil {
			head = this
			
		} else  {
			prev.Next = this
		}
		prev = this
	}
	if remainder ==1 {
		prev.Next = &ListNode{Val: 1}
	}
	return head
}
  
